### PR TITLE
Page detail feature/comments

### PR DESCRIPTION
**Pull request by brewost**
Add in comments to the film details page - make the params dynamic.

### CODE REVIEW

File: client/src/components/FilmComments.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  filmId: number;
+}
+
+const FilmComments: React.FC<FilmCommentsProps> = ({ filmId }) => {
+  const { loading, error, data } = useFilmCommentsQuery({
+    variables: { filmId },
+  });
+
+  if (loading) return <p>Loading comments...</p>;
+  if (error) return <p>Error loading comments: {error.message}</p>;
```
Comment (YOUNS28100) [inline client/src/components/FilmComments.tsx]: In French please 😁

File: client/src/components/FilmComments.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  const comments = data?.filmComments;
+
+  if (!comments || comments.length === 0) {
+    return <p>No comments available for this film.</p>;
```
Comment (YOUNS28100) [inline client/src/components/FilmComments.tsx]: In French too please 😁

**Review by YOUNS28100 — CHANGES_REQUESTED** [review]
Need to switch some text in english to french please

File: client/src/components/FilmComments.tsx (hunk 1 above)
Comment (brewost) [reply client/src/components/FilmComments.tsx]: in french🤣

File: client/src/components/FilmComments.tsx (hunk 2 above)
Comment (brewost) [reply client/src/components/FilmComments.tsx]: in french - sorry!😂

**Review by YOUNS28100 — DISMISSED** [review]

**Review by alexandreg67 — DISMISSED** [review]

**Review by Ludovicscelles — DISMISSED** [review]

File: api/src/resolvers/FilmResolver.ts [hunk 3]
```diff
@@ -81,4 +82,20 @@ export default class FilmResolver {
...
+  @Query(() => [UserComment], { nullable: true })
+  async filmComments(
+    @Arg("filmId", () => Int) filmId: number
+  ): Promise<UserComment[]> {
+    const film = await Film.findOne({ where: { id: filmId } });
+
+    if (!film) {
+      throw new Error("Film not found");
+    }
+    return UserComment.find({
```
Comment (jujuck) [inline api/src/resolvers/FilmResolver.ts]: ON va le faire en 2 temps
const userComment = await ...
return userComment

File: client/src/components/FilmComments.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+interface FilmCommentsProps {
+  filmId: number;
+}
+
+const FilmComments: React.FC<FilmCommentsProps> = ({ filmId }) => {
+  const { loading, error, data } = useFilmCommentsQuery({
+    variables: { filmId },
+  });
+
```
Comment (jujuck) [inline client/src/components/FilmComments.tsx:12]: Réorganise

File: client/src/components/FilmComments.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  }
+  const baseAvatarPath = "/avatar/";
+  return (
+    <section className="comments-section space-y-6">
+      {comments.map((comment, index) => {
+        const avatarPath = `${baseAvatarPath}${comment.user.avatar.image}`;
+
+        return (
+          <div
+            key={index}
```
Comment (jujuck) [inline client/src/components/FilmComments.tsx]: Pas d'index en key

File: client/src/pages/PageFilmDetail.tsx [hunk 6]
```diff
@@ -1,12 +1,16 @@
...
 import StarRating from "../components/NoteGlobale";
 import { useGetFilmByIdQuery } from "../generated/graphql-types";
+import FilmComments from "../components/FilmComments";
 
 const FilmDetail = () => {
-  // Hardcode the film ID
-  const hardcodedFilmId = 45; // Replace this with the desired film ID
+  const { id } = useParams<{ id: string }>();
   const basePosterUrl = "https://image.tmdb.org/t/p/original/"; // Base URL for TMDB posters
+  const filmId = id ? parseInt(id) : NaN;
```
Comment (jujuck) [inline client/src/pages/PageFilmDetail.tsx]: Pas besoin de ternaire, parseInt de undefined renvoie NaN

**Review by jujuck — CHANGES_REQUESTED** [review]

**Review by alexandreg67 — DISMISSED** [review]

**Review by Ludovicscelles — DISMISSED** [review]

**Review by jujuck — DISMISSED** [review]

**Review by jujuck — APPROVED** [review]

**Pull request merged**